frontend/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,29 @@
+import { ReactElement } from 'react';
+
+interface ButtonProps {
+  type?: 'button' | 'submit' | 'reset' | undefined;
```

Comment:
if you use ? after name (type?) it means that this value can be undefined, so there is no need to add  undefined to the type union.